core/arch/arm/plat-k3/drivers/ti_sci.c

```diff
@@ -82,80 +80,60 @@ static int ti_sci_setup_xfer(uint16_t msg_type, uint32_t msg_flags,
 }
 
 /**
- * ti_sci_get_response() - Receive response from mailbox channel
+ * ti_sci_do_xfer() - Do one transfer
  *
- * @xfer:	Transfer to initiate and wait for response
+ * @xfer: Transfer to initiate and wait for response
  *
  * Return: 0 if all goes well, else appropriate error message
  */
-static inline int ti_sci_get_response(struct ti_sci_xfer *xfer)
+static int ti_sci_do_xfer(struct ti_sci_xfer *xfer)
 {
-	struct k3_sec_proxy_msg *msg = &xfer->rx_message;
-	struct ti_sci_msg_hdr *hdr = NULL;
+	struct k3_sec_proxy_msg *txmsg = &xfer->tx_message;
+	struct k3_sec_proxy_msg *rxmsg = &xfer->rx_message;
+	struct ti_sci_msg_hdr *txhdr = (struct ti_sci_msg_hdr *)txmsg->buf;
+	struct ti_sci_msg_hdr *rxhdr = (struct ti_sci_msg_hdr *)rxmsg->buf;
+	static uint8_t message_sequence;
+	static struct mutex ti_sci_mutex_lock = MUTEX_INITIALIZER;
 	unsigned int retry = 5;
 	int ret = 0;
```

Comment:
There are still few inconsistencies in the return value type.
Here, `ret` should be of type `TEE_Result` (see `k3_sec_proxy_send()` and  `k3_sec_proxy_recv()`), but `ti_sci_do_xfer()` still returns an `int` value so there should be some conversion somewhere, of callers of `ti_sci_do_xfer()` should be changed to expect a `TEE_Result` value.

Reply:
I think this will have to be taken up in a separate PR @etienne-lms , this requires a cleanup across the full TISCI and secproxy layer to fix this..

Reply:
Yeah, I think this would require a proper cleanup so would do it together only in one shot.